extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/sql/resources/SqlStatementResource.java

```diff
@@ -0,0 +1,670 @@
+        if (!signature.isPresent()) {
+          return Response.ok().build();
+        }
+        Optional<List<Object>> results = SqlStatementResourceHelper.getResults(SqlStatementResourceHelper.getPayload(
+            contactOverlord(overlordClient.taskReportAsMap(queryId))));
+
+        return Response.ok((StreamingOutput) outputStream -> {
+          CountingOutputStream os = new CountingOutputStream(outputStream);
+
+          try (final ResultFormat.Writer writer = ResultFormat.OBJECT.createFormatter(os, jsonMapper)) {
+            List<ColumnNameAndTypes> rowSignature = signature.get();
+            writer.writeResponseStart();
+
+            for (long k = start; k < Math.min(last, results.get().size()); k++) {
+              writer.writeRowStart();
+              for (int i = 0; i < rowSignature.size(); i++) {
+                writer.writeRowField(
+                    rowSignature.get(i).getColName(),
+                    ((List) results.get().get(Math.toIntExact(k))).get(i)
+                );
+              }
+              writer.writeRowEnd();
+            }
+
+            writer.writeResponseEnd();
+          }
+          catch (Exception e) {
+            log.error(e, "Unable to stream results back for query[%s]", queryId);
+            throw new ISE(e, "Unable to stream results back for query[%s]", queryId);
+          }
+        }).build();
+
+      }
+    }
+    catch (DruidException e) {
+      return buildNonOkResponse(e);
+    }
+    catch (ForbiddenException e) {
+      log.debug("Got forbidden request for reason [%s]", e.getErrorMessage());
+      return buildNonOkResponse(
+          DruidException.forPersona(DruidException.Persona.USER)
+                        .ofCategory(DruidException.Category.FORBIDDEN)
+                        .build(Access.DEFAULT_ERROR_MESSAGE)
+      );
+    }
+    catch (Exception e) {
+      log.noStackTrace().warn(e, "Failed to handle query: %s", queryId);
+      return buildNonOkResponse(DruidException.forPersona(DruidException.Persona.DEVELOPER)
+                                              .ofCategory(DruidException.Category.UNCATEGORIZED)
+                                              .build(e, "Failed to handle query: [%s]", queryId));
+    }
+  }
+
+  /**
+   * Queries can be canceled while in any {@link SqlStatementState}. Canceling a query that has already completed will be a no-op.
+   *
+   * @param queryId queryId
+   * @param req     httpServletRequest
+   * @return HTTP 404 if the query ID does not exist,expired or originated by different user. HTTP 202 if the deletion
+   * request has been accepted.
+   */
+  @DELETE
+  @Path("/{id}")
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response deleteQuery(@PathParam("id") final String queryId, @Context final HttpServletRequest req)
+  {
+
+    try {
+      Access authResult = AuthorizationUtils.authorizeAllResourceActions(
```

Comment:
@abhishekagarwal87 I am following the pattern from https://github.com/apache/druid/blob/baa64e6d8a71b1d3816dea6144ca4322d0432e53/extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/sql/SqlTaskResource.java#L110. 

We might need to add a new method to AuthorizationUtilsto set the req header but that can be done as part of a follow up PR.

Reply:
https://github.com/apache/druid/pull/14878